test/ex_doc/language/elixir_test.exs

```diff
@@ -32,6 +32,20 @@ defmodule ExDoc.Language.ElixirTest do
       assert autolink_doc(~m"`PATH`") == ~m"`PATH`"
     end
 
+    test "invalid reference does not return backticks and emits warning" do
+      assert_invalid_ref_warning("`fakefunction`", fn ->
+        assert autolink_doc(~m"[text](`fakefunction`)") == ~m"text"
+      end)
+
+      assert_invalid_ref_warning("`t:supervisor.child_spec/0`", fn ->
+        assert autolink_doc(~m"[text](`t:supervisor.child_spec/0`)") == ~m"text"
+      end)
+
+      assert_invalid_ref_warning("`mix foobar`", fn ->
+        assert autolink_doc(~m"[text](`mix foobar`)") == ~m"text"
+      end)
```

Comment:
we're testing this scenario below on line 516-518. How are these different? warn/1 ensures warning in the form of `documentation references (...)`, assert_invalid_ref_warning/1 in the form of `found invalid reference ...`.  How come both tests are correct? Do we emit two warnings? Also note we tent to use `"` instead of `` ` `` in warning messages.

For the new warning my suggestion would be to have `documentation referenced "ref" but it is invalid` for consistency even though maybe it doesn't read perfectly for these types of warnings. This way we can use warn/1 test helper. Also nitpick

Could you also move these assertions to the one big warning test? I think it's helpful to see all possible warnings in one place.

Reply:
Hi @wojtekmach! Currently there were two warnings being emitted for those cases.

I agree we can update the warning message to be more consistent and emita single warning when we have invalid references.